app/controllers/admin/invoices_controller.rb

```diff
@@ -26,11 +26,15 @@ def index
     # GET /admin/invoices/search
     def search
       search_string = search_params[:search_string]
+      statuses_contains = params[:statuses_contains]
       @origin = search_string || search_params.dig(:order, :origin)
 
-      @invoices = search_scope(@origin).accessible_by(current_ability)
-                                       .order(orderable_array)
-                                       .page(1)
+      set_invoices_search_scope
+
+      return reform_array_to_paginate_array if statuses_contains.nil?
+
+      statuses_filter(statuses_contains)
+      reform_array_to_paginate_array
```

Comment:
I'd propose to rename this method for the sake of simplicity. Maybe something like `paginate_result` will do?